scripts/basic-workflow-tests.sh

```diff
@@ -186,6 +192,17 @@ test_train() {
     fi
 
     ilab model train "${TRAIN_ARGS[@]}"
+
+    if [ "$FULLTRAIN" -eq 1 ]; then
```

Comment:
When FULLTRAIN=1, we are running train twice.  Once with legacy set, but without 4-bit-quant, and then another time here in this code block you added.

Part of the issue is we've overridden the meaning of FULLTRAIN. It originally meant "don't use 4-bit-quant", though it's not a great name for that behavior.

A few options to resolve this:

1. Add a new option for using the training lib, and use that instead of FULLTRAIN
2. Change the legacy-without-4-bit-quant variant to have its own option and keep FULLTRAIN for this.
3. Leave it how it is where we run training twice, but add it to the list in an issue of follow-up cleanup tasks

Reply:
https://github.com/instructlab/instructlab/issues/1655